variables.tf

```diff
@@ -0,0 +1,37 @@
+# Variables
+
+variable "region" {
+  default = "us-east-2"
+}
+variable "cidr_vpc" {
+  description = "CIDR block for the VPC"
+  default     = "10.1.0.0/16"
+}
+variable "cidr_subnet" {
+  description = "CIDR block for the subnet"
+  default     = "10.1.0.0/24"
+}
+variable "cidr_subnet1" {
+  description = "CIDR block for the subnet1"
+  default     = "10.1.1.0/28"
+}
+variable "availability_zone" {
+  description = "availability zone to create subnet"
+  default     = "us-east-2a"
+}
+variable "public_key_path" {
+  description = "Public key path"
+  default     = "//~/.ssh/id_rsa.pub"
+}
+variable "instance_ami" {
+  description = "AMI for aws EC2 instance"
+  default     = "ami-0cf31d971a3ca20d6"
+}
```

Comment:
It it actually possible to deploy ec2 instance without having to hardcode ami id.
Please have a look here:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami